flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -129,6 +130,41 @@ class TaskManager(val connectionInfo: InstanceConnectionInfo,
     override def getValue: Double =
       ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
   })
+
+  // Preprocessing steps for registering cpuLoad
+  // fetch the method to get process CPU load
+  val getCPULoadMethod: Method = getMethodToFetchCPULoad()
+
+  // define the fetchCPULoad method as per the fetched getCPULoadMethod
+  val fetchCPULoad: (Any) => Double = if (getCPULoadMethod != null) {
+    (obj: Any) => getCPULoadMethod.invoke(obj).asInstanceOf[Double]
+  } else {
+    (obj: Any) => {
+      log.warning("getProcessCpuLoad method not available in the Operating System Bean" +
+        "implementation for this Java runtime environment",Thread.currentThread().getStackTrace)
+      -1
+    }
+  }
+
+  metricRegistry.register("cpuLoad", new Gauge[Double] {
+    override def getValue: Double = {
+      try{
+        val osMXBean = ManagementFactory.getOperatingSystemMXBean().
+          asInstanceOf[com.sun.management.OperatingSystemMXBean]
+        return fetchCPULoad(osMXBean)
+      } catch {
+        case t:Throwable => {
+          if (t.isInstanceOf[java.lang.ClassCastException]){
+            log.warning("Error casting to OperatingSystemMXBean",t)
```

Comment:
No, this error is not occurring for each heartbeat. This is in case there is a new error while using the getProcessCpuLoad method. I will remove the ClassCastException and make it Throwable so as to catch all types of errors.